### PR TITLE
Hacks for iOS demo

### PR DESCRIPTION
**Pull request by eseidel**
I've verified that this builds with 3.10.2.  I rebased onto shorebird/main before landing.

These are just the hacks needed for my current iOS demo.  This will still need more cleanup once the CLI/server sides are working.

Changes included:
* Split dart_sdk_revision and dart_sdk_git from dart_revision and dart_git since those affect other repos as well.
* Added a hacky path for loading symbols from an elf file on iOS.  Will need modification for production.
* Moved the ConfigureShorebird logic out into its own shorebird.cc file and shared between iOS and Android
* Added some hacks into iOS startup to use my hacky elf loader for loading in the dart snapshot instead of the default pathway.
* Turned on dart_force_simulator for iOS.

### CODE REVIEW

**Review by bryanoltman — APPROVED** [review]
¯\_(ツ)_/¯

**Comment by eseidel** [comment]
Waiting until the Dart bits land to land this, since this depends on the modified dart sdk.

File: shell/platform/darwin/ios/BUILD.gn
Comment (eseidel) [inline shell/platform/darwin/ios/BUILD.gn:203]: This should have been release.

**Pull request merged**
